### PR TITLE
feat(provisioning): provision MS Azure virtual machines (BCD-293)

### PR DESCRIPTION
**Pull request by tnguyen1**

### CODE REVIEW

**Review by alarreine — COMMENTED** [review]
I did not find any changes in scenarios.md, is that normal?

**Comment by tnguyen1** [comment]
Added scenario variables with commit 4d93b105b5bfac61394ee8d57a258a58c44c94a4

File: md/getting_started.md [hunk 1]
```diff
@@ -79,6 +79,10 @@ Then the `.boto` file has to be mounted from the control host into the BCD contr
 
     -v <host_path_to_.boto_file>:/home/bonita/.boto
 
+#### Azure Prerequisites
+
+If you consider deploying to Microsoft Azure cloud computing platform, first read the [Azure prerequisites tutorial](azure-prerequisites.md) which describes how to configure Azure for BCD.
+
 #### SSH key file
 
 BCD uses SSH to communicate with target machines. Therefore the BCD controller must have access to the related SSH private key file. For AWS, this is the private part of your [EC2 key pair](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-key-pairs.html). **This file must only be accessible from your user** (`chmod 600 <host_path_to_ssh_private_key_file>`).
```
Comment (JeremJR) [inline md/getting_started.md]: Could you please describe the Azure equivalent regarding ssh key configuration?

Comment (tnguyen1) [reply md/getting_started.md]: I added a quick note for Azure with this commit: 5baa7dfa8ce4940dd41e50fd81a3fceedb8bfb94

File: md/getting_started.md [hunk 2]
```diff
@@ -79,11 +79,18 @@ Then the `.boto` file has to be mounted from the control host into the BCD contr
...
+#### Azure Prerequisites
+
+If you consider deploying to Microsoft Azure cloud computing platform, first read the [Azure prerequisites tutorial](azure-prerequisites.md) which describes how to configure Azure for BCD.
+
 #### SSH key file
 
-BCD uses SSH to communicate with target machines. Therefore the BCD controller must have access to the related SSH private key file. For AWS, this is the private part of your [EC2 key pair](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-key-pairs.html). **This file must only be accessible from your user** (`chmod 600 <host_path_to_ssh_private_key_file>`).
+BCD uses SSH to communicate with target machines. Therefore the BCD controller must have access to the related SSH private key file.  
+For AWS, this is the private part of your [EC2 key pair](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-key-pairs.html) referenced in your AWS Console.  
+With Azure, you will have to create an SSH key pair using [this tutorial on Linux or macOS](https://docs.microsoft.com/en-us/azure/virtual-machines/linux/mac-create-ssh-keys) or [this tutorial on Windows](https://docs.microsoft.com/en-us/azure/virtual-machines/linux/ssh-from-windows).
```
Comment (JeremJR) [inline md/getting_started.md:90]: I'm not sure if it's here the better place to describe the differences but maybe that we should describe, here or under each prerequisites pages, a little be more the difference between AWS and Azure.
- with AWS, Terraform will only link an ec2 instance with the key name of a pubic key already uploaded
- with Azure we will automatically manage through Terraform the upload of the public key
it seems a different approach of Microsoft, if I understand well there is no equivalent to store the public key globally?
Apart of this PR we could discuss if we should try to use the same approach for AWS? https://www.terraform.io/docs/providers/aws/r/key_pair.html

File: md/scenarios.md [hunk 3]
```diff
@@ -43,13 +44,30 @@ The following variables are specific to AWS deployments ie. when `bcd_provider`
...
 | aws_load_balancer_instance_type | Y         | -       | EC2 instance type for the load balancer instance.                                                                                                                                                                                                                                                                                                                 | t2.micro                  |
 | aws_assumed_iam_role            | N         | -       |An IAM role can be assumed, so all requests are run as that role. This can be useful for connecting across different accounts, or to limit user access. Set this parameter if you want to assume an IAM role while deploying on AWS. It is useful in an AWS organization with IAM users and multiple AWS accounts. See [AWS Organizations](aws_organizations.md)   |arn:aws:iam::123456789012:role/BCD|
 
 
+## Azure provider variables
+
+The following variables are specific to Azure deployments ie. when `bcd_provider` is set to `azure`.
```
Comment (JeremJR) [inline md/scenarios.md:56]: Core variables should be updated accordingly

> Infrastructure provider. Choose between aws, azure and static_inventory

File: md/getting_started.md (hunk 2 above)
Comment (tnguyen1) [reply md/getting_started.md:90]: - It seems like you cannot create and store SSH key pairs on Azure. You have to provide the SSH public key (or username/password) whilc creating a virtual machine.
- With AWS, that would require permissions to import/create key pairs I guess? What happens if the key pair already exists? I'm not sure BCD should manage key create/deletion of key pairs.

File: md/scenarios.md (hunk 3 above)
Comment (tnguyen1) [reply md/scenarios.md:56]: Updated with commit 0aa9ae4

File: md/azure-prerequisites.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+       "AssignableScopes": [
+           "/subscriptions/XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX"
+       ],
+       "Description": "Read and write operations for BCD.",
+       "Name": "BCD Read/Write"
+   }
+   ```
+   **Note**: replace `XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX` with your actual subscription identifier. You may get your subscription identifier using Azure CLI with this command:  
+   `az account list --output table`  
+   Create your custom role using Azure CLI with this command:
```
Comment (alarreine) [inline md/azure-prerequisites.md]: I tried with a new azure account and I had to do `az login ` before.

**Review by alarreine — APPROVED** [review]

**Pull request merged**